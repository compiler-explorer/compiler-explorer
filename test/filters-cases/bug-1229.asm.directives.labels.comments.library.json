{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "_Z6myfuncv:                             # @_Z6myfuncv"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 9
      },
      "text": "        sub     rsp, 88"
    },
    {
      "labels": [],
      "source": {
        "column": 12,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/return_type.hpp",
        "line": 18
      },
      "text": "        mov     byte ptr [rsp + 32], 0"
    },
    {
      "labels": [],
      "source": {
        "column": 12,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/return_type.hpp",
        "line": 18
      },
      "text": "        mov     qword ptr [rsp + 24], 0"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     qword ptr [rsp + 40], offset .L.str"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     rax, qword ptr [rsp + 24]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     qword ptr [rsp + 48], rax"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     al, byte ptr [rsp + 32]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     byte ptr [rsp + 56], al"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     eax, dword ptr [rsp + 33]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     dword ptr [rsp + 57], eax"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        movzx   eax, word ptr [rsp + 37]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     word ptr [rsp + 61], ax"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     al, byte ptr [rsp + 39]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64
      },
      "text": "        mov     byte ptr [rsp + 63], al"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        mov     rax, qword ptr [rsp + 56]"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        mov     qword ptr [rsp + 16], rax"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        movups  xmm0, xmmword ptr [rsp + 40]"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        movups  xmmword ptr [rsp], xmm0"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        lea     rdi, [rsp + 64]"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        xor     esi, esi"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        mov     edx, offset .L.str"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        mov     ecx, offset .L.str+1"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        mov     r8d, offset .L.str+5"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267
      },
      "text": "        call    _ZN4ctre18evaluate_recursiveINS_13regex_resultsIPKcJEEES3_S3_Lm0ELm0EJNS_3anyEEJNS_10assert_endENS_8end_markENS_6acceptEEEET_mT0_SA_T1_S9_N4ctll4listIJNS_6repeatIXT2_EXT3_EJDpT4_EEEDpT5_EEE"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 11
      },
      "text": "        add     rsp, 88"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": null,
        "line": 11
      },
      "text": "        ret"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 298
      },
      "text": "        mov     rdi, rax"
    },
    {
      "labels": [
        {
          "name": "__clang_call_terminate",
          "range": {
            "endCol": 39,
            "startCol": 17
          }
        }
      ],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 298
      },
      "text": "        call    __clang_call_terminate"
    },
    {
      "labels": [],
      "source": null,
      "text": "__clang_call_terminate:                 # @__clang_call_terminate"
    },
    {
      "labels": [],
      "source": null,
      "text": "        push    rax"
    },
    {
      "labels": [],
      "source": null,
      "text": "        call    __cxa_begin_catch"
    },
    {
      "labels": [],
      "source": null,
      "text": "        call    _ZSt9terminatev"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L.str:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .asciz  \"hello\""
    },
    {
      "labels": [],
      "source": null,
      "text": ""
    }
  ],
  "labelDefinitions": {
    ".L.str": 34,
    "_Z6myfuncv": 1,
    "__clang_call_terminate": 30
  }
}
