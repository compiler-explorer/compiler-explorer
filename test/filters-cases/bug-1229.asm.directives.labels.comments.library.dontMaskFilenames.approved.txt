{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "_Z6myfuncv:                             # @_Z6myfuncv"
    },
    {
      "labels": [],
      "source": {
        "file": "/tmp/compiler-explorer-compiler1181120-2080-yfo6a1.y1o4e/example.cpp",
        "line": 9,
        "mainsource": true
      },
      "text": "        sub     rsp, 88"
    },
    {
      "labels": [],
      "source": {
        "column": 12,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/return_type.hpp",
        "line": 18,
        "mainsource": false
      },
      "text": "        mov     byte ptr [rsp + 32], 0"
    },
    {
      "labels": [],
      "source": {
        "column": 12,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/return_type.hpp",
        "line": 18,
        "mainsource": false
      },
      "text": "        mov     qword ptr [rsp + 24], 0"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     qword ptr [rsp + 40], offset .L.str"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     rax, qword ptr [rsp + 24]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     qword ptr [rsp + 48], rax"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     al, byte ptr [rsp + 32]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     byte ptr [rsp + 56], al"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     eax, dword ptr [rsp + 33]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     dword ptr [rsp + 57], eax"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        movzx   eax, word ptr [rsp + 37]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     word ptr [rsp + 61], ax"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     al, byte ptr [rsp + 39]"
    },
    {
      "labels": [],
      "source": {
        "column": 55,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 64,
        "mainsource": false
      },
      "text": "        mov     byte ptr [rsp + 63], al"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        mov     rax, qword ptr [rsp + 56]"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        mov     qword ptr [rsp + 16], rax"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        movups  xmm0, xmmword ptr [rsp + 40]"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        movups  xmmword ptr [rsp], xmm0"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        lea     rdi, [rsp + 64]"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        xor     esi, esi"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        mov     edx, offset .L.str"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        mov     ecx, offset .L.str+1"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        mov     r8d, offset .L.str+5"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 267,
        "mainsource": false
      },
      "text": "        call    _ZN4ctre18evaluate_recursiveINS_13regex_resultsIPKcJEEES3_S3_Lm0ELm0EJNS_3anyEEJNS_10assert_endENS_8end_markENS_6acceptEEEET_mT0_SA_T1_S9_N4ctll4listIJNS_6repeatIXT2_EXT3_EJDpT4_EEEDpT5_EEE"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": "/tmp/compiler-explorer-compiler1181120-2080-yfo6a1.y1o4e/example.cpp",
        "line": 11,
        "mainsource": true
      },
      "text": "        add     rsp, 88"
    },
    {
      "labels": [],
      "source": {
        "column": 1,
        "file": "/tmp/compiler-explorer-compiler1181120-2080-yfo6a1.y1o4e/example.cpp",
        "line": 11,
        "mainsource": true
      },
      "text": "        ret"
    },
    {
      "labels": [],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 298,
        "mainsource": false
      },
      "text": "        mov     rdi, rax"
    },
    {
      "labels": [
        {
          "name": "__clang_call_terminate",
          "range": {
            "endCol": 39,
            "startCol": 17
          }
        }
      ],
      "source": {
        "column": 9,
        "file": "/opt/compiler-explorer/libs/ctre/master/include/ctre/evaluation.hpp",
        "line": 298,
        "mainsource": false
      },
      "text": "        call    __clang_call_terminate"
    },
    {
      "labels": [],
      "source": null,
      "text": "__clang_call_terminate:                 # @__clang_call_terminate"
    },
    {
      "labels": [],
      "source": null,
      "text": "        push    rax"
    },
    {
      "labels": [],
      "source": null,
      "text": "        call    __cxa_begin_catch"
    },
    {
      "labels": [],
      "source": null,
      "text": "        call    _ZSt9terminatev"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L.str:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .asciz  \"hello\""
    },
    {
      "labels": [],
      "source": null,
      "text": ""
    }
  ],
  "labelDefinitions": {
    ".L.str": 34,
    "_Z6myfuncv": 1,
    "__clang_call_terminate": 30
  }
}
